lua/nvim-treesitter/refactor/highlight_definitions.lua

```diff
@@ -0,0 +1,75 @@
+-- This module highlights reference usages and the corresponding
+-- definition on cursor hold.
+
+local parsers = require'nvim-treesitter.parsers'
+local ts_utils = require'nvim-treesitter.ts_utils'
+local locals = require'nvim-treesitter.locals'
+local api = vim.api
+local cmd = api.nvim_command
+
+local M = {}
+
+local usage_namespace = api.nvim_create_namespace('nvim-treesitter-usages')
+
+function M.highlight_usages(bufnr)
+  M.clear_usage_highlights(bufnr)
+
+  local node_at_point = ts_utils.get_node_at_cursor()
+
+  if not node_at_point then return end
+
+  local def_node, scope = ts_utils.find_definition(node_at_point, bufnr)
+  local usages = ts_utils.find_usages(node_at_point, scope)
+
+  for _, usage_node in ipairs(usages) do
+    local start_row, start_col, _, end_col = usage_node:range()
+
+    if usage_node ~= node_at_point then
+      api.nvim_buf_add_highlight(
+        bufnr, 
+        usage_namespace, 
+        'Visual', 
```

Comment:
We should introduce a custom highlight linked to 'Visual' or something more subtle.